### PR TITLE
Adapt w.r.t. coq/coq#17322.

### PR DESCRIPTION
**Pull request by ppedrot**
This should be backwards compatible. The patch stops relying on a corner-case behaviour of specialize in presence of dependent unresolved bindings, which might change in the future.

### CODE REVIEW

**Comment by samuelgruetter** [comment]
Oh wow, so `specialize IHn with (1 := eq_refl)` creates an evar for one of the universally quantified variables of `IHn`, that's not the behavior I would have expected. Happy to see that this irregularity is about to be fixed!

**Comment by ppedrot** [comment]
@samuelgruetter Is there something to do to bump the tested branch, so that the Coq CI goes through, or is it automatically performed at regular intervals?

**Comment by samuelgruetter** [comment]
TL;DR: it should be fine now.

I manually bumped the riscv-coq submodule of bedrock2 yesterday, and usually, our GitHub Actions CI would build it, and then, nightly, the `tested` branch used by Coq's CI would get fast-forwarded to the latest commit on `master`. However, yesterday there was an error on GitHub Actions that I've never seen before: PRs (eg [this one](https://github.com/mit-plv/bedrock2/actions/runs/4318780874)) failed with "GitHub Actions has encountered an internal error when running your job", and pushes to `master` did not trigger any build and not create any tick or cross next to the commit. But I just manually triggered a [CI build](https://github.com/mit-plv/bedrock2/actions/runs/4325880808/jobs/7552599689) and the [fast-forward-tested-branch action](https://github.com/mit-plv/bedrock2/actions/runs/4326707165), which both succeeded (but without creating a green tick, though), so the `tested` branch should be up to date now. I hope this was just a temporary hickup on GitHub Actions.
